### PR TITLE
[CLI-2435] `confluent login` prompt by default

### PR DESCRIPTION
**Pull request by sgagniere**
Release Notes
-------------

Breaking Changes
- The `confluent login` command will no longer automatically log in using saved credentials in the keychain or configuration file

New Features
- PLACEHOLDER

Bug Fixes
- PLACEHOLDER

Checklist
---------
- [x] Leave this box unchecked if features are not yet available in production

What
----
Unless environment variables are specified, the `confluent login` command will always prompt for username (and password, when applicable).

I didn't remove the `--prompt` flag, so that users who have environment variables set can still bypass them.

References
----------

Test & Review
-------------

### CODE REVIEW

File: test/fixtures/output/login/help-onprem.golden [hunk 1]
```diff
@@ -40,7 +40,7 @@ Log in to Confluent Platform with a MDS URL and Certification Authority certific
 
   $ confluent login --url https://localhost:8090 --certificate-authority-path certs/my-cert.crt
 
-Log in to Confluent Platform with SSO and ignore any saved credentials.
+Log in to Confluent Platform with SSO and ignore login environment variables.
```
Comment (brianstrauch) [inline test/fixtures/output/login/help-onprem.golden:43]: We aren't ignoring any environment variables as far as I can tell. In fact, we require `CONFLUENT_PLATFORM_SSO` for this to work which we're definitely not ignoring.
```suggestion
Log in to Confluent Platform with SSO.
```

Comment (sgagniere) [reply test/fixtures/output/login/help-onprem.golden:43]: The reason I struggled with this one is that the `CONFLUENT_PLATFORM_SSO` env var is used to bypass the two credential env vars. And users without any env vars set don't need it either to log in with SSO.

So how about something like
```
Log in to Confluent Platform with SSO even if `CONFLUENT_PLATFORM_USERNAME` and `CONFLUENT_PLATFORM_PASSWORD` are set.
```

**Review by brianstrauch — APPROVED** [review]

File: internal/login/command.go [hunk 2]
```diff
@@ -60,7 +60,7 @@ func New(cfg *config.Config, prerunner pcmd.PreRunner, ccloudClientFactory pauth
 				Code: "confluent login --url https://localhost:8090 --certificate-authority-path certs/my-cert.crt",
 			},
 			examples.Example{
-				Text: "Log in to Confluent Platform with SSO and ignore any saved credentials.",
+				Text: "Log in to Confluent Platform with SSO and ignore login environment variables.",
```
Comment (sgagniere) [inline internal/login/command.go:63]:
```suggestion
				Text: "Log in to Confluent Platform with SSO even if `CONFLUENT_PLATFORM_USERNAME` and `CONFLUENT_PLATFORM_PASSWORD` are set.",
```

File: test/fixtures/output/login/help-onprem.golden (hunk 1 above)
Comment (sgagniere) [inline test/fixtures/output/login/help-onprem.golden:43]:
```suggestion
Log in to Confluent Platform with SSO even if `CONFLUENT_PLATFORM_USERNAME` and `CONFLUENT_PLATFORM_PASSWORD` are set.
```

File: test/fixtures/output/login/help.golden [hunk 3]
```diff
@@ -40,7 +40,7 @@ Log in to Confluent Platform with a MDS URL and Certification Authority certific
 
   $ confluent login --url https://localhost:8090 --certificate-authority-path certs/my-cert.crt
 
-Log in to Confluent Platform with SSO and ignore any saved credentials.
+Log in to Confluent Platform with SSO and ignore login environment variables.
```
Comment (sgagniere) [inline test/fixtures/output/login/help.golden:43]:
```suggestion
Log in to Confluent Platform with SSO even if `CONFLUENT_PLATFORM_USERNAME` and `CONFLUENT_PLATFORM_PASSWORD` are set.
```

**Pull request closed without merging**